internal/featuregates/features.go

```diff
@@ -0,0 +1,74 @@
+package featuregates
+
+/* Design aspects of implementing feature gates
+- feature gate is only for experimental features
+- Keep the feature gate as simple boolean.
+- Any config specific for a feature gate should be in it’s own configMap. This aligns with our current implementation of peer-pods and image generator feature
+- When we decide to make the feature as a stable feature, it should move to kataConfig.spec
+*/
+
+const (
+	FeatureGatesConfigMapName         = "osc-feature-gates"
+	ImageBasedDeployment              = "imageBasedDeployment"
+	ImageBasedDeploymentConfigMap     = "osc-feature-gate-image-deploy-config"
+	AdditionalRuntimeClasses          = "additionalRuntimeClasses"
+	AdditionalRuntimeClassesConfigMap = "osc-feature-gate-additional-rc-config"
+)
```

Comment:
Any reason for using "osc-feature-gate-" prefix ? I'm wondering if we couldn't just use the convention: FeatureName + '-config' for all features. That way we would avoid the need of most of this code here.

Reply:
See the response here - https://github.com/openshift/sandboxed-containers-operator/pull/394#discussion_r1574991356
The only reason was to make it easier for admins to know by just looking at the name when listing the configmaps.
May be I'll use `fg` instead of the full name.

Reply:
@beraldoleal @snir911 let me know if you would prefer to use FeatureName + '-config' and I'll change the code accordingly.

Reply:
I agree that we should easily identify what the ConfigMap is for. Either the "fg" prefix/suffix or a common namespace would help with that. But I would also try (if possible) to avoid having those hard coded for each feature and eventually remove the need of most of the code here.

So if possible, I would give a try with something like FeatureName + "-fg-cfg/-fg-config" suffix as a convention without the need of hardcode it.